move_base/src/move_base.cpp

```diff
@@ -946,6 +947,58 @@ namespace move_base {
     return hypot(p1.pose.position.x - p2.pose.position.x, p1.pose.position.y - p2.pose.position.y);
   }
 
+  bool MoveBase::getDistanceAndTimeEstimates(const tf::Stamped<tf::Pose>& poseTf,  std::vector<geometry_msgs::PoseStamped>& global_plan, double& distance, double& time)
+  {
+    // Get the current plan and pose.
+    if (global_plan.empty())
+    {
+      distance = 0;
+      time = 0;
+      return true;
+    }
+
+    geometry_msgs::PoseStamped pose;
+    tf::poseStampedTFToMsg(poseTf, pose);
+
+    Eigen::Vector2f pos2 = base_local_planner::poseStampedToVector(pose);
+    // Vectors for storage
+    Eigen::Vector2f p0 = Eigen::Vector2f::Zero();
+    Eigen::Vector2f p1 = Eigen::Vector2f::Zero();
+
+    // Other storage
+    double distance_from_robot = -1;
+    double minimum_distance = 0.5; // it needs to be at least this close
+
+    for (size_t k = 0; k < global_plan.size() - 1; ++k)
+    {
+      // Pull out the datas
+      p0 = base_local_planner::poseStampedToVector(global_plan[k]);
+      p1 = base_local_planner::poseStampedToVector(global_plan[k + 1]);
+      double segment_length = (p1 - p0).norm();
+      double dist_from_path = base_local_planner::distanceToLineSegment(pos2, p0, p1);
+
+      if (distance_from_robot < minimum_distance)
+      {
+        // reset distance
+        distance_from_robot = std::max(0.0, segment_length - base_local_planner::distanceAlongLineSegment(pos2, p0, p1));
+      }
+      else
+      {
+        distance_from_robot += segment_length;
+      }
+    }
+    if (distance_from_robot < 0)
+    {
+      // If the distance wasn't set, the robot is too far from the path.
+      // Use euclidean distance.
+      distance_from_robot = (pos2 - p1).norm();
+    }
+
+    distance = distance_from_robot/1.0;
```

Comment:
why divide by 1.0?  Is this the time calculation?  If so, comment what the 1.0 is from (arbitrary velocity).